### PR TITLE
Add a view with the list of checks for each release channel.

### LINKED ISSUE #62: Add an endpoint to list all the checks and their URL for a give release version.
Depending on the channel [we don't do exactly the same checks](https://github.com/nthomas-mozilla/release-pipelines).

For instance for the current checks that we have, we don't validate any security advisories for the beta or nightly channels.

1. We need to advertise that the check is not relevant for that version in the checks URL 
2. It might be even better to advertise the list of checks for a given version so that the delivery dashboard can discover them when we add new ones.

I propose to add a https://pollbot.dev.mozaws.net/v1/firefox/55.0.1 endpoint that would list the available checks like that:

```json
{
   "product": "firefox",
   "version": "55.0.1",
   "channel": "release",
   "checks": [
     {"url": "https://pollbot.dev.mozaws.net/v1/firefox/55.0.1/archive", "title": "Archives"},
     {"url": "https://pollbot.dev.mozaws.net/v1/firefox/55.0.1/bedrock/release-notes", "title": "Release notes"},
     {"url": "https://pollbot.dev.mozaws.net/v1/firefox/55.0.1/bedrock/security-advisories", "title": "Security Advisories"},
     {"url": "https://pollbot.dev.mozaws.net/v1/firefox/55.0.1/bedrock/download-links", "title": "Download links"},
     {"url": "https://pollbot.dev.mozaws.net/v1/firefox/55.0.1/product=-details", "title": "Product details"}
   ]
}
```

### PR DESCRIPTION
**Pull request by Natim**
Fixes #62

### CODE REVIEW

File: pollbot/views/release.py [hunk 1]
```diff
@@ -50,3 +54,63 @@ def status_response(task):
...
+
+    proto = request.headers.get('X-Forwarded-Proto', 'http')
+    host = request.headers['Host']
+
+    checks = []
+
+    for check_name, channels in CHECKS.items():
+        if channel in channels:
+            info = CHECKS_INFO[check_name].copy()
+            info['url'] = ("{}://{}" + info['url']).format(proto, host,
```
Comment (Natim) [inline pollbot/views/release.py]: Intermediate variable

File: pollbot/views/release.py [hunk 2]
```diff
@@ -50,3 +54,63 @@ def status_response(task):
...
+    "archive": {"url": "/v1/{product}/{version}/archive", "title": "Archive Release"},
+    "release-notes": {"url": "/v1/{product}/{version}/bedrock/release-notes",
+                      "title": "Release notes"},
+    "security-advisories": {"url": "/v1/{product}/{version}/bedrock/security-advisories",
+                            "title": "Security advisories"},
+    "download-links": {"url": "/v1/{product}/{version}/bedrock/download-links",
+                       "title": "Download links"},
+    "product-details": {"url": "/v1/{product}/{version}/product-details",
+                        "title": "Product details"},
+}
```
Comment (glasserc) [inline pollbot/views/release.py]: This seems kind of redundant with the route URLs that you feed to the router.

File: pollbot/tasks/archives.py [hunk 3]
```diff
@@ -17,10 +29,12 @@
...
-        else:
-            url = 'https://archive.mozilla.org/pub/{}/releases/{}/'.format(product, version)
-            async with session.get(url) as resp:
-                return resp.status != 404
+
+
+archives_date = partial(check_nightly_archives,
+                        'https://archive.mozilla.org/pub/{}/nightly/latest-date/')
+archives_date_l10n = partial(check_nightly_archives,
+                             'https://archive.mozilla.org/pub/{}/nightly/latest-date-l10n/')
```
Comment (glasserc) [inline pollbot/tasks/archives.py]: A few sentences in a module-level doc comment would be very helpful here. I guess we have to check whether a nightly appears in these places, and I guess these checks only make sense for Nightly?

File: pollbot/tasks/archives.py [hunk 4]
```diff
@@ -17,10 +29,12 @@
                                reverse=True)
                 last_release = get_version_from_filename(files[0][1])
                 return build_version_id(last_release) >= build_version_id(version)
```
Comment (glasserc) [inline pollbot/tasks/archives.py]: I know this is a tiny bit outside the scope of this PR, but I'm a bit confused here about what kind of "version" this is. Is this a Nightly version like `57.0a1`? Is it a buildhub nightly version like `2017-05-15-10-02-38_55-0a1`? What do we expect the frontend to pass?

File: pollbot/views/release.py (hunk 2 above)
Comment (Natim) [reply pollbot/views/release.py]: Yes I agree, I don't know if we can enhance that somehow

File: pollbot/tasks/archives.py (hunk 4 above)
Comment (Natim) [reply pollbot/tasks/archives.py]: This version 57.0a1, it is not related to buildhub versions

File: pollbot/tasks/archives.py (hunk 3 above)
Comment (Natim) [reply pollbot/tasks/archives.py]: Yes they do only make sense for nightly, they will return missing for other channels.
Also I need to update the OpenAPI spec file.

File: pollbot/views/release.py (hunk 2 above)
Comment (Natim) [reply pollbot/views/release.py]: http://aiohttp.readthedocs.io/en/stable/web.html#reverse-url-constructing-using-named-resources

File: pollbot/api.yaml [hunk 5]
```diff
@@ -194,6 +194,40 @@ paths:
       tags:
       - Status
 
+  /{product}/{version}/archive-date:
+    get:
+      summary: "checks if the product version archive is downloadable for nightly date."
```
Comment (glasserc) [inline pollbot/api.yaml]: I don't really understand what the word "date" here means. Should this be something like "checks if the product version package is available from the nightly latest-date archive"? Also, should there be periods at the end of these or not?

**Pull request merged**